### PR TITLE
🎖️ Integration specs for views and fix n+1 problems 🎖️

### PR DESCRIPTION
**Pull request by kambereBr**
## Project requirements

- ✅ No Linters error
- ✅ Used correct GitFlow.
- Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.

Before

n+1_before_

After

n+1_after_img

- Used Capybara to write integration tests for each view in the project (`User index page`, `user show page`, `User post index page`, `Post show page`)

⚠️⚠️⚠️  **About working alone** ⚠️⚠️⚠️

**My coding partner left the program before finishing all the tasks that he was supposed to do. Reason why you can see only two contributions in this milestone.**

### CODE REVIEW

File: spec/views/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    expect(page).to have_content('Post 2')
+    expect(page).to have_content('Post 3')
+  end
+
+  it "displays a button to view all of a user's posts" do
+    expect(page).to have_link('See all posts', href: user_posts_path(@user1))
+  end
+
+  it 'redirects to the user post index page when clicking on "See all posts"' do
+    click_link('See all posts')
```
Comment (MahmoudBakr23) [inline spec/views/user_show_spec.rb]: - [ ] In the user's show page testing file, please add the following test as per project requirements: "A test that redirects to the post's show page when you click on a user's post." 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello TEAM😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Great job on solving the N+1 query problems
- Well done on writing the required tests for each page so far
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/user_show_spec.rb (hunk 1 above)
Comment (kambereBr) [reply spec/views/user_show_spec.rb]: Dear @MahmoudBakr23, thanks for your review 👏

Comment (Reem-lab) [reply spec/views/user_show_spec.rb]: Hello @kambereBr I have a question you updated the user show but I think you didn't add a test that redirects to the post's show page right? to show a specific post for a user?

**Comment by kambereBr** [comment]
Dear @MahmoudBakr23, the issue fixed. Thanks

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @kambereBr  and @M-Anwar-Hussaini  :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kambereBr** [comment]
Dear @hiromataba, thanks for your review

**Pull request merged**